tools/flash-shelly.py

```diff
@@ -601,13 +565,51 @@ def device_scan(hosts, action, do_all, dry_run, silent_run, mode, exclude, versi
       deviceinfo.update_to_homekit(homekit_release_info)
     elif flashmode == 'stock':
       deviceinfo.update_to_stock(stock_release_info)
-
     if deviceinfo.fw_type == "homekit" and float(f"{parse_version(deviceinfo.info['version'])[0]}.{parse_version(deviceinfo.info['version'])[1]}") < 2.1:
       logger.info(f"{WHITE}Host: {NC}{deviceinfo.host}")
       logger.info(f"Version {deviceinfo.info['version']} is to old for this script,")
       logger.info(f"please update via the device webUI.\n")
-      continue
-    parse_info(deviceinfo, action, dry_run, silent_run, flashmode, exclude, version, hap_setup_code, requires_upgrade)
+    else:
+      parse_info(deviceinfo, action, dry_run, silent_run, flashmode, exclude, hap_setup_code, requires_upgrade)
+
+def device_scan(hosts, action, do_all, dry_run, silent_run, mode, exclude, version, variant, hap_setup_code):
+  logger.debug(f"\n{WHITE}device_scan{NC}")
+  if not do_all:
+    for host in hosts:
+      deviceinfo = Device(host)
+      deviceinfo.get_device_info()
+      if deviceinfo.fw_type is not None:
+        device = {'host': deviceinfo.host, 'wifi_ip': deviceinfo.wifi_ip, 'fw_type': deviceinfo.fw_type, 'device_url': deviceinfo.device_url, 'info': deviceinfo.info}
+        d_info = json.dumps(device, indent = 4)
+        logger.trace(f"Device Info: {d_info}")
+        probe_device(device, action, dry_run, silent_run, mode, exclude, version, variant, hap_setup_code)
+  else:
+    logger.info(f"{WHITE}Scanning for Shelly devices...{NC}")
+    zc = zeroconf.Zeroconf()
+    listener = MyListener()
+    browser = zeroconf.ServiceBrowser(zc, '_http._tcp.local.', listener)
+    zc.wait(100)
+    time.sleep(1)
+    d_queue = listener.queue
+    total_loop = 1
+    nod = 0
+    scan_finished = zc.done
+    while not scan_finished:
+      while d_queue.empty() and total_loop < 6: # do 6 loops of 2 seconds, if queue is still empty close scanner.
```

Comment:
use queue.get with timeout of, like, 20 seconds. when it times out, we're done

Reply:
sorry i dont understand, how does timeout help with looping queue ?, and detecting when queue is empty ?

Reply:
never mind, think I understand.